### PR TITLE
SelectMultiple new options on search

### PR DESCRIPTION
**Pull request by jcfilben**
#### What does this PR do?
Enables new options to be added to the SelectMultiple within the onSearch function

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?
Closes https://github.com/grommet/grommet/issues/7074

#### Do the grommet docs need to be updated?
yes
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: src/js/components/SelectMultiple/SelectMultiple.js [hunk 1]
```diff
@@ -152,6 +152,23 @@ const SelectMultiple = forwardRef(
       if (!search) setAllOptions(optionsProp);
     }, [optionsProp, search]);
 
+    useEffect(() => {
+      if (search && optionsProp && optionsProp.length > 0) {
+        const additionalOptions = [...allOptions];
+        optionsProp.forEach((i) =>
+          additionalOptions.some((j) => {
```
Comment (taysea) [inline src/js/components/SelectMultiple/SelectMultiple.js:159]: Syntax question, could this also be written as `!additionalOptions.some(...) && additionalOptions.push(i)` -- or is the `null` return needed specifically?

File: src/js/components/SelectMultiple/stories/SearchFetch.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+
+import { Box } from 'grommet';
+import { SelectMultiple } from '../SelectMultiple';
+
+const locationList = [
```
Comment (taysea) [inline src/js/components/SelectMultiple/stories/SearchFetch.js:6]: From offline discussion, I agree with not adding the storybook to the published set. Or if we do, build it out to be more representative of fetching from an API.

File: src/js/components/SelectMultiple/SelectMultiple.js [hunk 3]
```diff
@@ -152,6 +152,23 @@ const SelectMultiple = forwardRef(
...
+            if (typeof i === 'object')
```
Comment (taysea) [inline src/js/components/SelectMultiple/SelectMultiple.js:160]: Can this be rewritten to:

```
return typeof i === 'object' ? applyKey(i, valueKey) === applyKey(j, valueKey) : i === j;
```

Comment (taysea) [reply src/js/components/SelectMultiple/SelectMultiple.js:160]: Although if this feels less readable, i'm fine leaving the initial implementation

**Review by taysea — COMMENTED** [review]
Looking good, some minor comments!

File: src/js/components/SelectMultiple/SelectMultiple.js (hunk 1 above)
Comment (jcfilben) [reply src/js/components/SelectMultiple/SelectMultiple.js]: Good suggestion, the `null` return is not needed. Incorporated this suggestion in latest commit

File: src/js/components/SelectMultiple/SelectMultiple.js (hunk 3 above)
Comment (jcfilben) [reply src/js/components/SelectMultiple/SelectMultiple.js]: Incorporated this in the latest commit. I think you suggestion doesn't hurt readability

File: src/js/components/SelectMultiple/stories/SearchFetch.js (hunk 2 above)
Comment (jcfilben) [reply src/js/components/SelectMultiple/stories/SearchFetch.js:6]: Yeah I'm thinking we will remove this story before committing

**Review by taysea — APPROVED** [review]
I think this approach is reasonable for the time being. Once others approve, let's remove the storybook file.

Side note -- I was trying to think of an approach that might be a little more performant than mapping over the array with `.some` for each loop iteration, like potentially using a Set or some other structure. Although, since options can be an array of objects, the Set approach didn't exactly work. Not a blocker to moving forward with this, but just something I was thinking about.

**Review by MikeKingdom — APPROVED** [review]

**Pull request merged**